core/model/src/main/java/org/eclipse/sw360/antenna/model/artifact/facts/ArtifactFilename.java

```diff
@@ -120,11 +120,13 @@ public ArtifactFilename(String filename, String hash, String hashAlgorithm) {
                 .collect(Collectors.toList());
     }
 
-    public Optional<String> getBestFilenameGuess() {
-        return artifactFilenameEntries.stream()
-                .map(ArtifactFilenameEntry::getFilename)
-                .filter(Objects::nonNull)
-                .max(Comparator.comparing(String::length));
+    /*
+     * Current implementation uses a simple approach, the first entry in the set. This works in principle in use
+     * cases where there is only one entry in the model.
+     * Be aware of the limitation.
+     */
+    public Optional<ArtifactFilenameEntry> getBestFilenameEntryGuess() {
+        return artifactFilenameEntries.stream().findFirst();
```

Comment:
Why have you dropped the heuristics to use the the longest name? this even might return an entry which only contains a hash and a filename set to `null`